token/cli/src/command.rs

```diff
@@ -176,6 +177,51 @@ fn native_token_client_from_config(
     }
 }
 
+enum Pointer {
+    Metadata,
+    Group,
+    GroupMember,
+}
+impl Display for Pointer {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(match self {
+            Pointer::Metadata => "metadata pointer",
+            Pointer::Group => "group pointer",
+            Pointer::GroupMember => "member pointer",
+        })
+    }
+}
+impl Pointer {
+    pub async fn update_address(
```

Comment:
nitty nit: to be honest, I'm not a fan of small bits of encapsulation like this, since it forces me to wade all the way back to find this implementation. I think it would be easier to read to inline this directly in `command_update_pointer_address`. But that might be just me

Reply:
Sure, that's cool!